include/deal.II/meshworker/assemble_flags.h

```diff
@@ -0,0 +1,202 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2017 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii__mesh_worker_assemble_flags_h
+#define dealii__mesh_worker_assemble_flags_h
+
+
+#include <deal.II/base/config.h>
+#include <vector>
+
+
+DEAL_II_NAMESPACE_OPEN
+
+/*!@addtogroup MeshWorker */
+/*@{*/
+
+namespace MeshWorker
+{
+
+  /**
+   * The enum type given to the mesh_loop() function, telling that function which
+   * elements need to be assembled.
+   *
+   * You can select more than one flag by concatenation using the bitwise or
+   * `operator|(AssembleFlags,AssembleFlags)`.
+   *
+   * @author Luca Heltai, 2017.
+   */
+  enum AssembleFlags
+  {
+    //! No update
+    assemble_default = 0,
```

Comment:
we should discuss what the default should be (I think own_cells|faces_once or something)